seastar/src/spawn.rs

```diff
@@ -41,3 +41,47 @@ where
     .ok()?;
     Rc::try_unwrap(x).ok().unwrap().into_inner()
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[seastar::test]
+    async fn test_empty_spawn_void() {
+        assert_eq!(spawn(async move {}).await, Some(()));
+    }
+
+    #[seastar::test]
+    async fn test_chained_spawn_void() {
+        let res = spawn(async move {
+            let _ = spawn(async move {}).await;
+        })
+        .await;
+        assert!(matches!(res, Some(())));
+    }
+
+    #[seastar::test]
+    async fn test_spawn_int() {
+        let res = spawn(async move {
+            return 0;
+        })
+        .await;
+        assert!(matches!(res, Some(0)));
+    }
+
+    #[seastar::test]
+    async fn test_two_spawn_int_and_void() {
```

Comment:
I think we can replace this test with a "`test_spawn_void`" to avoid redundancy.

Reply:
Do you mean I should delete this test? I wrote it because there was a moment where a single spawn worked, but multiple spawns in a row didn't.

Reply:
What bugs me is the asymmetry of the `spawn_int` and `spawn_void` calls. I'd add a second `spawn_int` call in this test and then either keep it or split it into two.

Reply:
You can also think of testing those tests' logic with the `join!` macro - it'd make them more general over the order of the spawns (of course, in these tests, the tasks are kind of identical, so it doesn't matter, but that doesn't have to be the case).

Reply:
But there are two `spawn_int` calls. I still don't get what the problem is.